File: backend-springboot/src/main/java/com/splitwiseplus/model/Expense.java
package com.splitwiseplus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.splitwiseplus.util.ObjectIdDeserializer;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Document(collection = "expenses")
@Data
public class Expense {
    @Id
    @JsonProperty("_id")
    private String id;
    
    private String title;
    private Double amount;
    private String category;
    private String paidBy;
    private String paidByEmail;
    
    @JsonSerialize(using = ToStringSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private org.bson.types.ObjectId userId; // Reference to User ID
    
    private Date createdAt = new Date();
    private String notes = "";
    
    private List<SplitMember> splitWith;
    private Integer totalSplit = 1;
    
    @JsonSerialize(using = ToStringSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private org.bson.types.ObjectId groupId;

    @Data
    public static class SplitMember {
        private String name;
        private String email;
        private Double amount;
    }
}

File: backend/models/Expense.js
const mongoose = require("mongoose")

const expenseSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  amount: {
    type: Number,
    required: true,
  },
  category: {
    type: String,
    required: true,
  },
  paidBy: {
    type: String,
    required: true,
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  notes: {
    type: String,
    default: ""
  },
  splitWith: [
    {
      name: String,
      email: String,
      amount: Number
    }
  ],
  totalSplit: {
    type: Number,
    default: 1
  },
  groupId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Group",
    default: null
  }
})

module.exports = mongoose.model("Expense", expenseSchema)
